### PR TITLE
[WIP] feat: add lambda auth docs

### PR DESCRIPTION
**Pull request by iartemiev**
_Issue #, if available:_

_Description of changes:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Amazon Cognito Identity Pools allows you to use credentials from AWS IAM in your
 
 If you are using a 3rd party OIDC provider you will need to configure it and manage the details of token refreshes yourself. 
 
+### Lambda Authorizer
+
+You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you API, please see the Settings section of the AppSync console. You will need to manage the details of token refreshes in your application code yourself. 
+
```
Comment (diegocstn) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:24]: should we also add a code snippet of an example of Lambda here?

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 2]
```diff
@@ -158,6 +162,26 @@ If you are using Cognito's user pool as the authorization type, this will by def
 
 <inline-fragment platform="ios" src="~/lib/graphqlapi/fragments/ios/authz/21_oidc.md"></inline-fragment>
 <inline-fragment platform="android" src="~/lib/graphqlapi/fragments/android/authz/21_oidc.md"></inline-fragment>
+### Lambda Authorizer
+
+Update the `amplifyconfiguration.json` file and code snippet as follows:
```
Comment (diegocstn) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:167]: nit: can we maybe rephrase this to point out that devs only need to update the `authorizationType` key?

**Review by diegocstn — APPROVED** [review]

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md (hunk 1 above)
Comment (iartemiev) [reply docs/lib/graphqlapi/fragments/native_common/authz/common.md:24]: My plan was to link to the AppSync docs for Lambda auth here (once they become publicly available) and those would give customers more info on configuring a Lambda authorizer. We'll be adding CLI support in the near future, which would generate a basic authorizer for them.

File: docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+## Configure Lambda Auth Mode with DataStore
+
+You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you AppSync API, please see the Settings section of the AppSync console.
```
Comment (renebrandel) [inline docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md:3]:
```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you AppSync API, please see the Settings section of the AppSync console.
```

File: docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+## Configure Lambda Auth Mode with DataStore
```
Comment (renebrandel) [inline docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md:1]:
```suggestion
## Configure custom authorization logic with AWS Lambda
```

File: docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you AppSync API, please see the Settings section of the AppSync console.
+
+You will need to manage the details of token refreshes in your application code yourself. 
+
+When using Lambda authorizers with DataStore here's how you can specify a function for handling token refresh:
```
Comment (renebrandel) [inline docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md:7]:
```suggestion
Here's how you can specify a function for handling token refresh when using Lambda as an authorization mode with DataStore:
```

File: docs/lib/graphqlapi/fragments/js/authz.md [hunk 6]
```diff
@@ -32,6 +33,37 @@ When using __AWS_IAM__ for public API access, unauthenticated logins must be ena
 
 </amplify-callout>
 
+### Lambda
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/js/authz.md:36]:
```suggestion
### AWS Lambda
```

File: docs/lib/graphqlapi/fragments/js/authz.md [hunk 7]
```diff
@@ -32,6 +33,37 @@ When using __AWS_IAM__ for public API access, unauthenticated logins must be ena
...
+
+You can implement your own API authorization logic using an AWS Lambda function. Lambda authorizers can be configured via the settings section in the AppSync console.
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/js/authz.md:38]:
```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda as an authorization mode for you AppSync API, go to the **Settings** section of the **AppSync console**.
```

File: docs/lib/graphqlapi/fragments/js/authz.md [hunk 8]
```diff
@@ -32,6 +33,37 @@ When using __AWS_IAM__ for public API access, unauthenticated logins must be ena
...
+
+If you are using a Lambda authorizer with your AppSync API, you will need to pass an authentication token with each API request and will need to manage token refresh in your application.
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/js/authz.md:40]:
```suggestion
If you are using a Lambda as an authorization mode with your AppSync API, you will need to pass an authentication token with each API request and will need to manage token refresh in your application.
```

File: docs/lib/graphqlapi/fragments/js/authz.md [hunk 9]
```diff
@@ -136,3 +168,29 @@ const client = new AWSAppSyncClient({
   },
 });
 ```
+
+### Lambda
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/js/authz.md:172]:
```suggestion
### AWS Lambda
```

File: docs/lib/graphqlapi/fragments/js/authz.md [hunk 10]
```diff
@@ -136,3 +168,29 @@ const client = new AWSAppSyncClient({
...
+You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you API, please see the Settings section of the AppSync console.
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/js/authz.md:173]:
```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda as an authorization mode for you AppSync API, go to the **Settings** section of the **AppSync console**.
```

File: docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
+## Configure custom authorization logic with AWS Lambda
+
+You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda as an authorization mode for you AppSync API, go to the **Settings** section of the **AppSync console**.
```
Comment (renebrandel) [inline docs/lib/datastore/fragments/js/setup-auth-rules/20_function-auth-snippet.md:3]:
```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda function as an authorization mode for you AppSync API, go to the **Settings** section of the **AppSync console**.
```

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 12]
```diff
@@ -18,6 +18,10 @@ Amazon Cognito Identity Pools allows you to use credentials from AWS IAM in your
 
 If you are using a 3rd party OIDC provider you will need to configure it and manage the details of token refreshes yourself. 
 
+### Lambda
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:21]:
```suggestion
### AWS Lambda
```

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 13]
```diff
@@ -18,6 +18,10 @@ Amazon Cognito Identity Pools allows you to use credentials from AWS IAM in your
...
+
+You can implement your own API authorization logic using an AWS Lambda function. To add a Lambda authorizer as an authentication mode for you API, please see the Settings section of the AppSync console. You will need to manage the details of token refreshes in your application code yourself.
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:23]:
```suggestion
You can implement your own custom API authorization logic using an AWS Lambda function. To add a Lambda function as an authorization mode for your AppSync API, go to the **Settings** section of the **AppSync console**.

You will need to manage the details of token refreshes in your application code yourself. 
```

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 14]
```diff
@@ -158,6 +162,26 @@ If you are using Cognito's user pool as the authorization type, this will by def
 
 <inline-fragment platform="ios" src="~/lib/graphqlapi/fragments/ios/authz/21_oidc.md"></inline-fragment>
 <inline-fragment platform="android" src="~/lib/graphqlapi/fragments/android/authz/21_oidc.md"></inline-fragment>
+### Lambda
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:165]:
```suggestion
### AWS Lambda
```

File: docs/lib/graphqlapi/fragments/native_common/authz/common.md [hunk 15]
```diff
@@ -158,6 +162,26 @@ If you are using Cognito's user pool as the authorization type, this will by def
...
+
+Update the `authorizationType` value in the `amplifyconfiguration.json` file and code snippet as follows:
```
Comment (renebrandel) [inline docs/lib/graphqlapi/fragments/native_common/authz/common.md:169]:
```suggestion
Amplify CLI does not currently allow you to configure Lambda as an authorization mode for your GraphQL API. To add a Lambda function as an authorization mode for your AppSync API, go to the **Settings** section of the **AppSync console**. Then, update the `authorizationType` value in the `amplifyconfiguration.json` file and code snippet as follows:
```

**Review by renebrandel — APPROVED** [review]
